src/tests/server.ts

```diff
@@ -854,51 +956,60 @@ describe('Subscribe', () => {
             },
           }),
         );
+      } else {
+        dMessage.resolve(message);
       }
-      return message;
+      invocationCounter++;
     });
     client.onmessage = onMessageFn;
-    await wait(10);
+
+    await dServerReceivedSubscribe.promise;
 
     pubsub.publish('boughtBananas', {
       boughtBananas: {
         name: 'john',
       },
     });
-    await wait(10);
 
-    expect(onMessageFn.mock.results[1].value).toEqual({
+    let value = await dMessage.promise;
+
+    expect(value).toEqual({
       id: '1',
       type: MessageType.Next,
       payload: { data: { boughtBananas: { name: 'john' } } },
     });
 
+    dMessage = createDeferred();
+
     // complete
     client.send(
       stringifyMessage<MessageType.Complete>({
         id: '1',
         type: MessageType.Complete,
       }),
     );
-    await wait(10);
+
+    value = await dMessage.promise;
 
     // confirm complete
-    expect(onMessageFn).toHaveLastReturnedWith({
+    expect(value).toEqual({
       id: '1',
       type: MessageType.Complete,
     });
 
-    pubsub.publish('boughtBananas', new Error('Something weird happened!'));
-    await wait(10);
+    // TODO: if we use something like a PushPull publish we can better confirm nobody is subscribing anymore
```

Comment:
See https://github.com/n1ru4l/graphql-live-queries/blob/a4ce5ac727a7c44fd205b824d678db7717abd220/packages/in-memory-live-query-store/src/PushPullAsyncIterableIterator.ts#L1

We could simply check `pullQueue.length` 🤔

But since the idea of the test is that it would show that no more messages should be dispatched it is also hard to test this without waiting 😔